src/lv_gpu/lv_gpu_stm32_dma2d.c

```diff
@@ -64,6 +56,21 @@ static DMA2D_HandleTypeDef hdma2d;
  *   GLOBAL FUNCTIONS
  **********************/
 
+/**
+ * Turn on the peripheral and set output color mode, this only needs to be done once
+ */
+void lv_gpu_stm32_dma2d_init(void)
+{
+    /* Enable DMA2D clock */
+    RCC->AHB1ENR |= RCC_AHB1ENR_DMA2DEN;
```

Comment:
Is a `DMB` needed after this on Cortex-M7? Not 100% sure.

Reply:
Not sure what you mean by DMB?

Reply:
It's a [data memory barrier](http://infocenter.arm.com/help/index.jsp?topic=/com.arm.doc.faqs/ka14041.html). That page is for ARM in general but I know the instruction exists and is sometimes needed on Cortex-M7.

Reply:
Got you. I don't think it does as that area of memory is type Device which guarantees order.

In table B3-1 of the ARMv7-M manual (https://static.docs.arm.com/ddi0403/eb/DDI0403E_B_armv7m_arm.pdf), it states that memory block 0x40000000-0x5FFFFFFF is memory type Device. In http://infocenter.arm.com/help/index.jsp?topic=/com.arm.doc.dui0646a/CIHGEIID.html it states.

Device and Strongly-ordered
    The processor preserves transaction order relative to other transactions to Device or Strongly-ordered memory.

As those two lines of code produce 1 register read, followed by a register write, then register read all to Device type memory, the order is guaranteed.